TextureHolder.cpp

```diff
@@ -0,0 +1,45 @@
+#include "TextureHolder.h"
+#include <assert.h>
+
+using namespace sf;
+
+TextureHolder* TextureHolder::m_s_Instance = NULL;
+
+TextureHolder::TextureHolder()
+{
+	assert(m_s_Instance == NULL);
+	m_s_Instance = this;
+}
+

```

Comment:
I promised to give my thoughts on another way to make the singleton instance. But the thing is, it is not a pattern I like that much and there are different approaches (depending on how you need to use it).

If you want to try it out still, I would recommend going the direction I had shown in the other comment (with a new class defined in this cpp file).

In general singletons can be nice, but I would recommend not relying on them too much. When your project grows, the singleton can cause more headaches than anything else, since now you are managing the lifetime and state of an object that everyone can access and it is not super clear when things should happen and who is responsible for them!